client/state/purchases/actions.js

```diff
@@ -56,14 +56,9 @@ export const cancelPrivacyProtection = purchaseId => dispatch => {
 		} );
 };
 
-export const clearPurchases = () => {
-	// TODO: we should call this olark method elsewhere, so that this
-	// action creator doesn't have side effects.
-	olark.updateOlarkGroupAndEligibility();
-
-	return {
-		type: PURCHASES_REMOVE,
-	};
+export const clearPurchases = () => dispatch => {
+	dispatch( { type: PURCHASES_REMOVE } );
+	dispatch( requestHappychatEligibility() );
```

Comment:
Is the TODO message about side effects not relevant anymore?

Reply:
Yes, I believe it's not relevant any more.

Before this patch, the Olark function was called in the action creator -- you didn't even need to dispatch the action to execute it (!). Very hacky. Now it's a classic thunk that dispatches two Redux actions. Nothing unusual any more. Converting the request to data layer would be even better, but I didn't have any energy left to do that :)